src/messages.ts

```diff
@@ -264,6 +271,12 @@ export function handleMessage(this: any, msg: { data: string }): void {
       })
     }
 
+    if (event && event.categoryCode === 'simulate') {
+      newState.contractCall = event.transaction.contractCall
+      delete newState.dispatchTimestamp
+      simulations$.next(newState)
+    }
```

Comment:
```suggestion
    if (event && event.categoryCode === 'simulate') {
      newState.contractCall = event.transaction.contractCall
      delete newState.dispatchTimestamp
      simulations$.next(newState)
      return
    }
```
Could we also do an early return here to prevent unnecessary work?